### PR TITLE
feat: Supply account id for functions and ef

### LINKED ISSUE #5825: Supply Account ID to Edge Functions and V2 Serverless Functions
In https://github.com/netlify/cli/pull/5804, we didn't implement providing `account.id` to Lambda. It's also not available in Edge Functions right now.

We should provide `account.id` to both Edge Functions and Serverless Functions in the CLI.

### PR DESCRIPTION
**Pull request by jrwhitmer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #5825 

This work was todo after https://github.com/netlify/cli/pull/5804. For functions and edge functions we were not passing the `account id` through to Lambda. In this PR, we have added the `account id`. A note that we were not able to write a test for this, but we were able to test it locally.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

**Comment by jrwhitmer** [comment]
apologies for the repeated commit message, I have never pushed here before, and I got in a small disagreement with `prettier`

**Comment by Skn0tt** [comment]
The CI check is a bit misleading - it wasn't failing because the Commit message had the wrong format, but because the PR title had the wrong format (the PR title is used for the squashed commit when merging). I added the `feat:` type, and now it's passing.

File: src/functions-templates/typescript/set-res-header/{{name}}.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import type { Context } from "https://edge.netlify.com";
 
 export default async (request: Request, context: Context) => {
-  const response = await context.next();
-  response.headers.set("X-Your-Custom-Header", "A custom value");
-  return response;
+  const newURL = new URL("/sale", request.url);
```
Comment (Skn0tt) [inline src/functions-templates/typescript/set-res-header/{{name}}.ts:4]: I believe this is unrelated to the change, could you remove this?

File: src/functions-templates/typescript/transform-response/{{name}}.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
 import { Context } from "https://edge.netlify.com";
 
 export default async (request: Request, context: Context) => {
-  const url = new URL(request.url);
+  const newURL = new URL(request.url);
 
   // Look for the query parameter, and return if we don't find it
-  if (url.searchParams.get("method") !== "transform") {
+  if (newURL.searchParams.get("method") !== "transform") {
```
Comment (Skn0tt) [inline src/functions-templates/typescript/transform-response/{{name}}.ts:7]: same here

File: src/lib/edge-functions/proxy.mjs [hunk 3]
```diff
@@ -106,6 +114,8 @@ export const initializeProxy = async ({
     // Setting header with geolocation and site info.
     req.headers[headers.Geo] = JSON.stringify(geoLocation)
     req.headers[headers.Site] = createSiteInfoHeader(siteInfo)
+    console.log(accountId)
```
Comment (Skn0tt) [inline src/lib/edge-functions/proxy.mjs:117]:
```suggestion
```

File: src/utils/proxy.mjs [hunk 4]
```diff
@@ -60,6 +60,7 @@ const decompressResponseBody = async function (body, contentEncoding = '') {
 }
 
 const formatEdgeFunctionError = (errorBuffer, acceptsHtml) => {
+  console.log(errorBuffer.toString())
```
Comment (Skn0tt) [inline src/utils/proxy.mjs:63]:
```suggestion
```

**Review by Skn0tt — COMMENTED** [review]
Looks good! Left some notes on parts we should remove from the PR.

File: src/lib/edge-functions/proxy.mjs (hunk 3 above)
Comment (jrwhitmer) [reply src/lib/edge-functions/proxy.mjs:117]: Whoops.

**Review by sarahetter — DISMISSED** [review]
LGTM!

**Review by sarahetter — APPROVED** [review]
